spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AppConfigurationPropertySourceLocator.java

```diff
@@ -137,28 +137,35 @@ private void addPropertySource(CompositePropertySource composite, ConfigStore st
         // There is only one Feature Set for all AppConfigurationPropertySources
         FeatureSet featureSet = new FeatureSet();
 
+        List<AppConfigurationPropertySource> sourceList = new ArrayList<AppConfigurationPropertySource>();
+
         // Reverse in order to add Profile specific properties earlier, and last profile
         // comes first
         Collections.reverse(contexts);
         for (String sourceContext : contexts) {
             try {
-                List<AppConfigurationPropertySource> sourceList = create(sourceContext, store, storeContextsMap,
-                        initFeatures, featureSet);
-                sourceList.forEach(composite::addPropertySource);
+                sourceList.addAll(create(sourceContext, store, storeContextsMap, initFeatures, featureSet));
+
                 LOGGER.debug("PropertySource context [{}] is added.", sourceContext);
             } catch (Exception e) {
                 if (store.isFailFast() || !startup) {
                     LOGGER.error(
                             "Fail fast is set and there was an error reading configuration from Azure App "
-                                    + "Configuration Service for " + sourceContext);
+                                    + "Configuration store " + store.getEndpoint()
+                                    + ". The configuration starting with " + sourceContext + " failed to load.");
                     ReflectionUtils.rethrowRuntimeException(e);
                 } else {
-                    LOGGER.warn("Unable to load configuration from Azure AppConfiguration Service for " + sourceContext,
+                    LOGGER.warn(
+                            "Unable to load configuration from Azure AppConfiguration store " + store.getEndpoint()
+                                    + ". The configurations starting with " + sourceContext + "failed to load.",
                             e);
                     StateHolder.setLoadState(store.getEndpoint(), false);
                 }
+                // If anything breaks we skip out on loading the rest of the store.
+                return;
```

Comment:
If we return here, we'll not be able to load the store.

Reply:
That is the idea. If a user uses both /application/ and /application_name/. One can load while the other does not. This causes confusion and error.

Reply:
That is the idea. If a user uses both /application/ and /application_name/. One can load while the other does not. This causes confusion and error.